runtime/str.go

```diff
@@ -819,11 +819,17 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
 		if matches[7] != "%" && valueIndex >= len(values.elems) {
 			return nil, f.RaiseType(TypeErrorType, "not enough arguments for format string")
 		}
-		if matches[1] != "" {
-			return nil, f.RaiseType(NotImplementedErrorType, "conversion flags not yet supported")
-		}
-		if matches[2] != "" || matches[4] != "" {
-			return nil, f.RaiseType(NotImplementedErrorType, "field width not yet supported")
+		if (matches[1] == "" || matches[1] == "0") && matches[2] != "*" && (matches[7] == "d" || matches[7] == "x" || matches[7] == "X") {
```

Comment:
You can invert this condition so that you don't need the empty if clause:

```
if (matches[1] != "" && matches[1] != "0") || matches[2] == "*" || (matches[7] != "d" && matches[7] != "x" && matches[7] != "X") {
```